mergify_engine/tests/functional/actions/test_queue.py

```diff
@@ -3591,6 +3591,103 @@ async def test_create_pull_basic(self):
+
+        p1, _ = await self.create_pr()
+        p2, _ = await self.create_pr()
+        p3, _ = await self.create_pr()
+        p4, _ = await self.create_pr()
+        p5, _ = await self.create_pr()
+
+        ctxt = context.Context(self.repository_ctxt, p1)
+        q = await merge_train.Train.from_context(ctxt)
+        base_sha = await q.get_base_sha()
+
+        queue_config = rules.QueueConfig(
+            priority=0,
+            speculative_checks=5,
+            batch_size=1,
+            batch_max_wait_time=datetime.timedelta(seconds=0),
+            allow_inplace_checks=True,
+            disallow_checks_interruption_from_queues=[],
+            checks_timeout=None,
+            draft_bot_account=None,
+        )
+
+        config = queue.PullQueueConfig(
+            name="foo",
+            strict_method="merge",
+            update_method="merge",
+            priority=0,
+            effective_priority=0,
+            bot_account=None,
+            update_bot_account=None,
+            queue_config=queue_config,
+        )
+
+        car1 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p1["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p1["number"], config, date.utcnow())],
+            [p1["number"]],
+            base_sha,
+        )
+        q._cars.append(car1)
+
+        car2 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p2["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p2["number"], config, date.utcnow())],
+            [p1["number"]],
+            base_sha,
+        )
+
+        car3 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p3["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p3["number"], config, date.utcnow())],
+            [p2["number"]],
+            base_sha,
+        )
+
+        car4 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p4["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p4["number"], config, date.utcnow())],
+            [p2["number"]],
+            base_sha,
+        )
+        q._cars.append(car4)
+
+        car5 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p5["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p5["number"], config, date.utcnow())],
+            [p3["number"]],
+            base_sha,
+        )
+        q._cars.append(car5)
+
+        queue_rule = rules.QueueRule(
+            name="foo",
+            conditions=conditions.QueueRuleConditions([]),
+            config=queue_config,
+        )
+
+        await car1.create_pull(queue_rule)
+        await car2.create_pull(queue_rule)
+        await car3.create_pull(queue_rule)
+        await car4.create_pull(queue_rule)
+
+        assert 6 == len(await self.get_branches())
+        assert 3 == len(q._cars)
+
```

Comment:
You also need to create branches unrelated to the main branch and to the train cars.

(setup_repo() has an attribute called test_branches that you can leverage for this.)

Reply:
Indeed, forgot to create those ! Gonna change this !